.pre-commit-config.yaml

```diff
@@ -1,11 +1,17 @@
 repos:
-    - repo: https://github.com/PyCQA/isort
-      rev: 5.12.0
+    - repo: https://github.com/charliermarsh/ruff-pre-commit
+      rev: v0.0.254
       hooks:
-          - id: isort
+        - id: ruff
+    - repo: https://github.com/Lucas-C/pre-commit-hooks
+      rev: v1.4.2
+      hooks:
+        - id: forbid-tabs
+          exclude: Makefile|docs/Makefile|\.bat$|\.cc$|\.h
```

Comment:
Tabs are the worst, but if we exclude everything there's almost no point in this hook.

Reply:
I excluded the Makefiles because if can cause issues if we remove the tabs: https://github.com/Lucas-C/pre-commit-hooks/issues/22 And I was being overly cautious and didn't want to modify the c++ files in this pr because we're not re-compiling them and checking everything still works